library/farmosUtil/farmosUtil.js

```diff
@@ -1644,6 +1644,117 @@ export async function getPlantAsset(plantAssetId) {
   return results.data[0]; // only one asset with the plantAssetId.
 }
 
+/**
+ * Get the plant assets at the specified location.
+ *
+ * @param {string} locationName the name of the location.
+ * @param {string[]} [checkedBeds=[]] the beds to look for.
+ * @param {boolean} [isInGround=true] look for plants in the ground.
+ * @param {boolean} [isInTrays=true] look for plants in trays.
+ * @return {Object} the plant assets at the specified location.
+ * @throws {Error} if unable to fetch the plant assets.
+ *
+ * @category Plant
+ */
+export async function getPlantAssets(
+  locationName,
+  checkedBeds = [],
+  isInTrays = true,
+  isInGround = true
+) {
+  const farm = await getFarmOSInstance();
+  let bedNameToAssetMap = null;
+  let greenhouseNameToAssetMap = null;
+  let fieldNameToAssetMap = null;
+  let fieldAsset = null;
+  let greenhouseAsset = null;
+  let locationAsset = null;
+
+  // If both isInTrays and isInGround are false, return null
+  if (!isInTrays && !isInGround) {
+    return [];
+  } else if (!isInGround) {
+    // not in ground (i.e. only in trays)
+    // location must be a greenhouse
+    greenhouseNameToAssetMap = await getGreenhouseNameToAssetMap();
+  } else {
+    // not sure about location:
+    greenhouseNameToAssetMap = await getGreenhouseNameToAssetMap();
+    fieldNameToAssetMap = await getFieldNameToAssetMap();
+  }
+
+  if (checkedBeds.length > 0) {
+    bedNameToAssetMap = await getBedNameToAssetMap();
+  }
+
+  // Find the field asset that matches the location name
+  if (fieldNameToAssetMap) {
+    fieldAsset = fieldNameToAssetMap.get(locationName);
+  }
+  if (greenhouseNameToAssetMap) {
+    greenhouseAsset = greenhouseNameToAssetMap.get(locationName);
+  }
+
+  if (greenhouseAsset) {
+    locationAsset = greenhouseAsset;
+  } else if (fieldAsset) {
+    locationAsset = fieldAsset;
+  } else {
+    return [];
+  }
+
+  // Fetch all plant assets
+  const plantAssets = await farm.asset.fetch({
```

Comment:
Add `limit: Infinity` to this filter so that we get all of the relevant plants.  Otherwise, we will only get one page.  See `getLogCategories` or something similar for an example.